crypto/evp/evp_pbe.c

```diff
@@ -25,41 +27,42 @@ struct evp_pbe_st {
     int cipher_nid;
     int md_nid;
     EVP_PBE_KEYGEN *keygen;
```

Comment:
Do you need the keygen one at all here.. since it is just keygenex(... NULL, NULL)

Reply:
I kept it in for backward compatibility with the mechanism which allows PBEs to be added at runtime. I'm not sure how widely used EVP_PBE_alg_add()/EVP_PBE_alg_add_type() are but removing them would simplify the code.
Come to think of it I'm not sure they would even work now with most of the code calling keygen_ex.